smart-actions/src/main/java/org/smartdata/actions/hdfs/MoveFileAction.java

```diff
@@ -58,27 +59,24 @@ public void init(Map<String, String> args) {
     this.storagePolicy = args.get(STORAGE_POLICY);
   }
 
-  protected void execute() {
+  @Override
+  protected void execute() throws ActionException {
     logOut.println("Action starts at "
         + (new Date(System.currentTimeMillis())).toString() + " : "
         + fileName + " -> " + storagePolicy.toString());
     try {
       dfsClient.setStoragePolicy(fileName, storagePolicy);
     } catch (Exception e) {
-      actionStatus.end();
-      actionStatus.setSuccessful(false);
-      throw new RuntimeException(e);
+      throw new ActionException(e);
```

Comment:
How about:
throw new ActionException("Failed to setStoragePolicy for file {} with storage policy {}", e);